File: Python3/1189. Maximum Number of Balloons.py
# 40ms, 68.15%; 13.8MB, 100.00%
from collections import Counter

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        instance, tmp = Counter('balloon'), Counter(text)
        res = len(text)
        for v in instance:
            res = min(res, tmp[v] // instance[v])
        return res

File: 1001 - 1200/1189 Maximum Number of Balloons.cpp
// 0ms, 100.00%; 8.7MB, 100.00%
class Solution
{
public:
    int maxNumberOfBalloons(const string &text)
    {
        int instance[26] = {}, tmp[26] = {};
        instance[0] = instance[1] = instance[13] = 1;
        instance[11] = instance[14] = 2;
        for (auto c : text) tmp[c - 97] += 1;
        int res = text.size();
        for (int i = 0; i < 26; ++i) {
            if (instance[i] == 0) continue;
            res = min(res, tmp[i] / instance[i]);
        }
        return res;
    }
};
